### PR TITLE
feat(cmd/init): change default module name to project name

### LINKED ISSUE #3277: Propose Change: Default Module Name to Project Name
### Is your feature request related to a problem? Please describe.

_No response_

### Describe the solution you'd like

The solution I propose is to modify the Go Wails project templates to automatically generate a `go.mod` file with at least the project name as the module name, instead of the default placeholder "changeme".

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Twacqwq**
# Description
Change the default module name `changeme` to the project name.

```bash
$ wails init -n myproject -t vue-ts
```

Fixes #3277 

## Type of change
  
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
  
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration using `wails doctor`.

- [ ] Windows
- [ ] macOS
- [x] Linux
  
### Wails
Version         | v2.8.0
Package Manager | pacman

### System
|-|-|
|-|-|
| OS           | Arch Linux                          |
| Version      | TEMPLATE_VERSION_ID                 |
| ID           | arch                                |
| Go Version   | go1.21.5                            |
| Platform     | linux                               |
| Architecture | amd64                               |
| CPU          | 13th Gen Intel(R) Core(TM) i7-13700 |
| GPU          | Unknown                             |
| Memory       | 15GB                                |

### Dependencies
|-|-|
|-|-|
| Dependency | Package Name | Status    | Version     |
| *docker    | docker       | Installed | 1:25.0.3-1  |
| gcc        | gcc          | Installed | 13.2.1-5    |
| libgtk-3   | gtk3         | Installed | 1:3.24.41-1 |
| libwebkit  | webkit2gtk   | Installed | 2.42.5-2    |
| npm        | npm          | Installed | 10.5.0-1    |
| pkg-config | pkgconf      | Installed | 2.1.0-2     |

# Checklist:

- [x] I have updated `website/src/pages/changelog.mdx` with details of this PR
- [x] My code follows the general coding style of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Summary by CodeRabbit

- **New Features**
    - Introduced an option to automatically update the default module name to match the project name during project initialization.
    - Added functionality to update the module name in the `go.mod` file based on user input.
    - Added a new command line option `-m` to change the `go.mod` module name to the project name.
    - Changed the functionality to create a project with the project's name instead of the default name.

### CODE REVIEW

File: v2/internal/gomod/gomod.go [hunk 1]
```diff
@@ -112,3 +112,13 @@ func SyncGoVersion(goModText []byte, goVersion string) ([]byte, bool, error) {
 
 	return goModText, true, nil
 }
+
+func UpdateGoModuleName(goModText []byte, newModuleName string) ([]byte, error) {
+	file, err := modfile.Parse("", goModText, nil)
```
Comment (leaanthony) [inline v2/internal/gomod/gomod.go:117]: No doubt this works just fine, but I'm wondering if `go mod edit -module <name>` would be more forward compatible?

**Comment by leaanthony** [comment]
This is a great start!
Please could you add an entry to the changelog located at `website/src/pages/changelog.mdx`?
We will also need to update the CLI reference documentation in `website/docs/reference/cli.mdx`.

File: v2/internal/gomod/gomod.go (hunk 1 above)
Comment (Twacqwq) [reply v2/internal/gomod/gomod.go:117]: You're right! This is indeed a lack of consideration on my part. Compared with calling the `golang.org/x/mod/modfile` library to parse `go.mod`, the latter is more forward-compatible using commands. I will update its implementation and update the PR.

**Comment by Twacqwq** [comment]
> This is a great start! Please could you add an entry to the changelog located at `website/src/pages/changelog.mdx`? We will also need to update the CLI reference documentation in `website/docs/reference/cli.mdx`.

of course I'd love to do that.

**Comment by tmclane** [comment]
I would also recommend that the default name of the module be set to the name of the directory created for the project if not overridden with the `-m` option.

**Comment by Twacqwq** [comment]
@leaanthony If I remove the `-m` parameter, is it okay to replace the module name of `go.mod` with the project name by default?

**Comment by leaanthony** [comment]
Sure.

**Comment by Twacqwq** [comment]
@leaanthony Ready for review.

**Review by leaanthony — APPROVED** [review]

**Pull request merged**